File: VTI_Galaxy_BE/src/main/java/demo/services/interfaceClass/RoomService.java
package demo.services.interfaceClass;

import demo.modal.dto.RoomDto;
import demo.modal.request.RoomRequest;

import java.util.List;

public interface RoomService {
    // get
    List<RoomDto> getAllRooms();
    RoomDto getRoomById(int id);
    List<RoomDto> getByGalaxy(int galaxyId);
    RoomDto getRoomByShowTime(int movieId, int galaxyId, String startTime);

    // post
    RoomDto createRoom(RoomRequest request);

    // put
    RoomDto updateRoom(RoomRequest request, int id);

    // delete (closed)
    RoomDto closeRoom(int id);
}

File: VTI_Glaxy_FE/src/services/roomService.js
import axiosClient from "./axiosClient";

const roomService = {
  fetchRooms() {
    return axiosClient.get("/getRooms");
  },
  fetchRoomById(roomId) {
    return axiosClient.get(`/getRoomById?roomId=${roomId}`);
  },
  fetchRoomByGalaxy(galaxyId) {
    return axiosClient.get(`/getRoomByGalaxy?galaxyId=${galaxyId}`);
  },
  fetchRoomByShowTime(movieId, galaxyId, time){
    return axiosClient.get(`/getRoomByShowTime?movieId=${movieId}&galaxyId=${galaxyId}&time=${time}`)
  },
  createRoom(roomRequest) {
    return axiosClient.post("/postRoom", roomRequest);
  },
  updateRoom(roomId, roomRequest) {
    return axiosClient.put(`/putRoom?roomId=${roomId}`, roomRequest);
  },
  deleteRoom(roomId) {
    return axiosClient.delete(`/deleteRoom?roomId=${roomId}`);
  },
};
export default roomService;
